pkg/proxy/util/utils.go

```diff
@@ -441,3 +441,25 @@ func GetClusterIPByFamily(ipFamily v1.IPFamily, service *v1.Service) string {
 
 	return ""
 }
+
+// FilterIncorrectLoadBalancerIngress filters out the ingresses with an IP version different from the given one
+func FilterIncorrectLoadBalancerIngress(ingresses []v1.LoadBalancerIngress, ipFamily v1.IPFamily) ([]v1.LoadBalancerIngress, []v1.LoadBalancerIngress) {
+	var validIngresses []v1.LoadBalancerIngress
+	var invalidIngresses []v1.LoadBalancerIngress
+
+	for _, ing := range ingresses {
+		correctIP := MapIPsByIPFamily([]string{ing.IP})[ipFamily]
```

Comment:
This feels like going out of the way to use the existing API, but it doesn't really fit.  You're going to waste time mapping by IP when you KNOW there's just one IP.  I think it would be cleaner to iterate the input list, map by family, then return that map.